esphome/components/xdb401/xdb401.cpp

```diff
@@ -0,0 +1,179 @@
+#include "xdb401.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace xdb401 {
+
+static const char *const TAG = "xdb401";
+
+static const uint8_t REG_PRESSURE = 0x06;
+static const uint8_t REG_TEMPERATURE = 0x09;
+static const uint8_t REG_MAKE_MEASURE = 0x30;
+static const uint8_t CMD_MAKE_MEASURE = 0x0A;
+static const uint8_t MASK_MEASURE_READY = 0x08;
+static const float CONVERT_PRESSURE = (float) 0x800000;
+static const float SCALE_PRESSURE = 1.0e6;
+static const float SCALE_TEMPERATURE = 100.0;
+
+static const int CHECK_DELAY = 5u;
+static const int CHECK_ATTEMPTS = 6u;
+static const int MARK_FAIL_AFTER = 5;
+
+void XDB401Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up XDB401...");
+
+  float temperature;
+  float pressure;
+  i2c::ErrorCode err_code = this->read_(temperature, pressure);
+  if (err_code != i2c::ERROR_OK) {
+    // this->mark_failed("I2C Communication Failed...");  // Need latest dev
+    this->mark_failed();
+    ESP_LOGE(TAG, "I2C Communication Failed...");
```

Comment:
https://github.com/esphome/esphome/blob/3470305d9d41837a8412eee0b4f6fdf247e6beed/esphome/core/log_const_en.h#L3
```suggestion
    this->mark_failed(ESP_LOG_MSG_COMM_FAIL);
```